pulp_smash/tests/rpm/api_v2/test_rsync_distributor.py

```diff
@@ -204,16 +205,53 @@ def make_user(self, cfg):
             being targeted.
         :returns: A ``(username, private_key)`` tuple.
         """
-        sudo = '' if utils.is_root(cfg) else 'sudo '
         creator = _make_user(cfg)
         username = next(creator)
-        self.addCleanup(
-            cli.Client(cfg).run,
-            (sudo + 'userdel --remove {}').format(username).split()
-        )
+        self.addCleanup(self.delete_user, cfg, username)
         private_key = next(creator)
         return (username, private_key)
 
+    @staticmethod
+    def delete_user(cfg, username):
+        """Delete a user.
+
+        The Pulp rsync distributor has a habit of leaving (idle?) SSH sessions
+        open even after publishing a repository. When executed, this function
+        will:
+
+        1. Poll the process list until all processes belonging to ``username``
+           have died, or raise a ``unittest.SkipTest`` exception if the time
+           limit is exceeded.
+        2. Delete ``username``.
+        """
+        sudo = () if utils.is_root(cfg) else ('sudo',)
+        client = cli.Client(cfg)
+
+        # values are arbitrary
+        iter_time = 2  # seconds
+        iter_limit = 15  # unitless
+
+        # Wait for user's processes to die.
+        cmd = sudo + ('ps', '-wwo', 'args', '--user', username, '--no-headers')
+        i = 0
+        while i <= iter_limit:
+            try:
+                user_processes = client.run(cmd).stdout.splitlines()
+            except exceptions.CalledProcessError:
```

Comment:
This command fails if `ps` finds no processes associated with the named user. Take this example from my system:

```console
$ ps -wwo args --user foo --no-headers ; echo $?
error: user name does not exist

Usage:
 ps [options]

 Try 'ps --help <simple|list|output|threads|misc|all>'
  or 'ps --help <s|l|o|t|m|a>'
 for additional help text.

For more details see ps(1).
1
# useradd foo
$ ps -wwo args --user foo --no-headers ; echo $?
1
```

As you can see in the latter case, the user "foo" exists, and `ps` returns 1 because there are no processes associated with that user.